src/cache.ts

```diff
@@ -83,31 +83,34 @@ function findPackageManager(id: string): PackageManager {
 /**
  * A function that generates a cache key to use.
  * Format of the generated key will be "${{ platform }}-${{ id }}-${{ fileHash }}"".
- * If there is no file matched to {@link PackageManager.path}, the generated key ends with a dash (-).
  * @see {@link https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key|spec of cache key}
  */
-async function computeCacheKey(packageManager: PackageManager) {
-  const hash = await glob.hashFiles(packageManager.pattern.join('\n'));
-  return `${CACHE_KEY_PREFIX}-${process.env['RUNNER_OS']}-${packageManager.id}-${hash}`;
+async function computeCacheKey(
+  packageManager: PackageManager,
+  cacheDependencyPath?: string
+) {
+  const pattern = cacheDependencyPath
+    ? cacheDependencyPath.trim().split('\n')
+    : packageManager.pattern;
+  const fileHash = await glob.hashFiles(pattern.join('\n'));
+  if (!fileHash) {
+    throw new Error(
+      `No file in ${process.cwd()} matched to [${pattern}], make sure you have checked out the target repository`
```

Comment:
Let's convert this to warning, as this logic changes the way how action worked previously, by that I mean this comment:
`If there is no file matched to {@link PackageManager.path}, the generated key ends with a dash (-).`

Reply:
Previously, the error was thrown [here](https://github.com/actions/setup-java/pull/499/files#diff-d027d3b9853e6bedddf63bd1d8fa67c69b4c1ff2b947685108f315e4e831cdbdL104). So I think the action works as same as before.